release/pkg/assets/archives.go

```diff
@@ -0,0 +1,142 @@
+package assets
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+
+	assettypes "github.com/aws/eks-anywhere/release/pkg/assets/types"
+	"github.com/aws/eks-anywhere/release/pkg/filereader"
+	releasetypes "github.com/aws/eks-anywhere/release/pkg/types"
+	artifactutils "github.com/aws/eks-anywhere/release/pkg/util/artifacts"
+	sliceutils "github.com/aws/eks-anywhere/release/pkg/util/slices"
+)
+
+func getArchiveAssets(rc *releasetypes.ReleaseConfig, archive *assettypes.Archive, projectPath, gitTag, eksDReleaseChannel, eksDReleaseNumber, kubeVersion string) (*releasetypes.ArchiveArtifact, error) {
+	os := "linux"
+	arch := "amd64"
+	var sourceS3Key string
+	var sourceS3Prefix string
+	var releaseS3Path string
+	var releaseName string
+	sourcedFromBranch := rc.BuildRepoBranchName
+	latestPath := artifactutils.GetLatestUploadDestination(sourcedFromBranch)
+
+	if archive.Name == "eks-distro" {
```

Comment:
I would vote to try to avoid this kind of thing
Instead of using branching, could we codify this in the archive definition?
Like for example, using something like `ArchiveS3KeyGenerator`, `ArchiveS3PathGenerator`... (they can be closures or interfaces)
Those could just be part of each `Archive` declaration and then we can do `archive.getS3Path()`

The same thing applies to all the if/else for `rc.DevRelease` and `rc.ReleaseEnvironment`, could we avoid them with dependency injection?